src/utils/assetManagers/assetManagers.js

```diff
@@ -233,6 +234,38 @@ export function insertDomain({ domain }, callback) {
   promise.catch(e => callback(e))
 }
 
+/**
+ * Retrieve EOD Books
+ * @param {object} params - object of parameters:
+ * @param {Domain} params.AMID 
+ * @param {Domain} params.bookID
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is the inserted Domain instance. Omit to return promise.
+ * @returns {Promise|null} If no callback supplied, returns a promise that resolves with the inserted Domain instance.
+ */
+export function retrieveEODBooks({ AMId, bookID }, callback) {
+  const params = {
+    AMaaSClass: 'assetManagerEODBooks',
+    AMId,
+    bookID
+  }
+  let promise = retrieveData(params).then(result => {
```

Comment:
Does this return a single object or an array or either? If it returns either, you will need to do a check to see if it's an Array and then map the result to an Array of EODBooks. You can refer to the `retrieve` function in utils/assets to see what I mean.